### PR TITLE
fix(components): [el-transfer] can't find clearQuery

### LINKED ISSUE #5431: [Bug Report] Uncaught TypeError: Cannot set properties of null (setting 'query')
### Element Plus version
1.3.0-beta.5

### OS/Browsers version
Chrome 版本 97.0.4692.71（正式版本） (x86_64)

### Vue version
3.2.16

### Reproduction Link
https://codesandbox.io/s/element-plus-spa-forked-9tdpi?file=/src/App.vue

### Steps to reproduce
1、点击Clear按钮，执行clearQuery方法

### What is Expected?
清除左侧输入框里的内容

### What is actually happening?
Uncaught TypeError: Cannot set properties of null (setting 'query')

### PR DESCRIPTION
**Pull request by btea**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

fix #5431

### CODE REVIEW

File: packages/components/transfer/src/index.vue [hunk 1]
```diff
@@ -183,14 +185,14 @@ export default defineComponent({
       emit
     )
 
-    const leftPanel = ref(null)
-    const rightPanel = ref(null)
+    const leftPanel = ref<TransferType | null>(null)
+    const rightPanel = ref<TransferType | null>(null)
```
Comment (emojiiii) [inline packages/components/transfer/src/index.vue:189]: may be `const left Panel = ref<TransferType>()` will be better

**Review by emojiiii — APPROVED** [review]

File: packages/components/transfer/src/index.vue (hunk 1 above)
Comment (btea) [reply packages/components/transfer/src/index.vue]: Updated

File: packages/components/transfer/src/index.vue [hunk 2]
```diff
@@ -183,14 +185,14 @@ export default defineComponent({
...
+    const leftPanel = ref<TransferType>()
+    const rightPanel = ref<TransferType>()
 
     const clearQuery = (which: 'left' | 'right') => {
       if (which === 'left') {
-        leftPanel.value.query = ''
+        leftPanel.value!.query = ''
       } else if (which === 'right') {
-        rightPanel.value.query = ''
+        rightPanel.value!.query = ''
```
Comment (JeremyWuuuuu) [inline packages/components/transfer/src/index.vue:195]: Can we use `switch` in this case?

Comment (JeremyWuuuuu) [reply packages/components/transfer/src/index.vue:195]: I would also like to ask you to add a test against this, so we can assure this code won't be wrong again.

**Review by jw-foss — COMMENTED** [review]
I left 2 comments for you to consider, let me know if you have any questions.

File: packages/components/transfer/src/index.vue (hunk 2 above)
Comment (btea) [reply packages/components/transfer/src/index.vue:195]: > Can we use `switch` in this case?  

ok.

> I would also like to ask you to add a test against this, so we can assure this code won't be wrong again.  

Good advice.

**Review by sxzz — APPROVED** [review]

**Pull request merged**